### DISCUSSION TITLE
Change to version numbering (UIs and BE components)

### DISCUSSION THREAD

**Discussion by severinbeauvais** [bcgov/lear#1374] [Ideas]
Currently, the Entity UI versions are 4.2.4/3.1.2/0.6.24, which are pretty meaningless.

I propose to change the UI versions to `yyyy.mm.dd` -- updated in each PR (as we do now), starting with PRs in the new year.

Any strong feelings either way? Or other ideas?

**Comment by severinbeauvais** [comment]
@thorwolpert wrote:
> Personally, I’d like something to track the deployment of “the service” and the useful component list.
so for the api meta end-point, I’d like more info
> - release: ?? - agreed it is some arbitrary, but recognizable value
> - deployment: prefer a date here
> - main service version number
> - main service commit hash
> - major libraries: eg business-schema version

**Reply by thorwolpert** [reply]
thanks!

**Comment by pwei1018** [comment]
Think about using a manifest file to store those information and integrate with CD flow.

**Reply by severinbeauvais** [reply]
That's a great idea! Especially if we choose something like "today" (yyyy.mm.dd), the CD script could auto-populate that.

Does the version in package.json become obsolete then?

**Reply by pwei1018** [reply]
> That's a great idea! Especially if we choose something like "today" (yyyy.mm.dd), the CD script could auto-populate that.
> 
> Does the version in package.json become obsolete then?

CD flow can also change the package.json version number automatically.

**Comment by pwei1018** [comment]
I have a example 'version file' that update by CD flow. 
https://github.com/bcgov/bcregistry-sre/blob/main/vault-service/devops/versions.yaml

Makefile has the update command.
https://github.com/bcgov/bcregistry-sre/blob/main/vault-service/Makefile
